### PR TITLE
Add documentation to Values property in TypedConstant struct

### LINKED ISSUE #72046: Warning AD0001 : Analyzer '' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
**Version Used**: 
Microsoft.CodeAnalysis.CSharp 4.3.0

This repository will throw the exception when building:
https://github.com/thomhurst/AD0001.Replication

Basically I'm passing in an argument of `null` to a `params object?[]` parameter in an attribute.

The source code for the analyzer that is throwing the exception is here:
```csharp
using System.Collections.Immutable;
using System.Linq;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;
using Microsoft.CodeAnalysis.Diagnostics;

namespace TUnit.Analyzers;

[DiagnosticAnalyzer(LanguageNames.CSharp)]
public class DataDrivenTestArgumentsAnalyzer : ConcurrentDiagnosticAnalyzer
{
    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
        ImmutableArray.Create(Rules.InvalidDataAssertion, Rules.NoDataProvidedAssertion);

    public override void InitializeInternal(AnalysisContext context)
    { 
        context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.MethodDeclaration);
    }
    
    private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
    { 
        if (context.Node is not MethodDeclarationSyntax methodDeclarationSyntax)
        {
            return;
        }

        if (context.SemanticModel.GetDeclaredSymbol(methodDeclarationSyntax)
            is not { } methodSymbol)
        {
            return;
        }

        var attributes = methodSymbol.GetAttributes();
        
        foreach (var dataDrivenTestAttribute in attributes.Where(x => x.AttributeClass?.ToDisplayString(DisplayFormats.FullyQualifiedNonGenericWithGlobalPrefix)
                                                == "global::TUnit.Core.DataDrivenTestAttribute"))
        {
            CheckAttributeAgainstMethod(context, methodSymbol, dataDrivenTestAttribute);
        }
    }

    private void CheckAttributeAgainstMethod(SyntaxNodeAnalysisContext context, IMethodSymbol methodSymbol,
        AttributeData dataDrivenTestAttribute)
    {
        if (!dataDrivenTestAttribute.ConstructorArguments.Any())
        {
            context.ReportDiagnostic(
                Diagnostic.Create(Rules.NoDataProvidedAssertion,
                    dataDrivenTestAttribute.ApplicationSyntaxReference?.GetSyntax().GetLocation())
            );
            return;
        }

        var methodParameterTypes = methodSymbol.Parameters.Select(x => x.Type).ToList();
        var attributeTypesPassedIn = dataDrivenTestAttribute.ConstructorArguments.First().Values.Select(x => x.Type).ToList();
        
        if (methodParameterTypes.Count != attributeTypesPassedIn.Count)
        {
            context.ReportDiagnostic(
                Diagnostic.Create(Rules.InvalidDataAssertion,
                    dataDrivenTestAttribute.ApplicationSyntaxReference?.GetSyntax().GetLocation(),
                    string.Join(", ", attributeTypesPassedIn.Select(x => x?.ToDisplayString())),
                    string.Join(", ", methodParameterTypes.Select(x => x?.ToDisplayString())))
            );
            return;
        }
        
        for (var i = 0; i < methodParameterTypes.Count; i++)
        {
            var methodParameterType = methodParameterTypes[i];
            var attributeArgumentType = attributeTypesPassedIn[i];
            
            if (IsEnumAndInteger(methodParameterType, attributeArgumentType))
            {
                continue;
            }
            
            if (!context.Compilation.HasImplicitConversion(attributeArgumentType, methodParameterType))
            {
                context.ReportDiagnostic(
                    Diagnostic.Create(Rules.InvalidDataAssertion,
                        dataDrivenTestAttribute.ApplicationSyntaxReference?.GetSyntax().GetLocation(),
                        attributeArgumentType?.ToDisplayString(),
                        methodParameterType?.ToDisplayString())
                );
            }
        }
    }

    private bool IsEnumAndInteger(ITypeSymbol type1, ITypeSymbol? type2)
    {
        if (type1.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat) == "int")
        {
            return type2?.TypeKind == TypeKind.Enum;
        }
        
        if (type2?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat) == "int")
        {
            return type1.TypeKind == TypeKind.Enum;
        }

        return false;
    }
}
```

This code has NO nullability warnings and I am also not suppressing any nullability warnings. Yet it is apparently throwing a NullReferenceException.

My best guess is that this line is the issue? But I'm really not sure.
```csharp
dataDrivenTestAttribute.ConstructorArguments.First().Values.Select(x => x.Type).ToList();
```

### PR DESCRIPTION
**Pull request by GiovanniBraconi**
Fixes #72046

### CODE REVIEW

File: src/Compilers/Core/Portable/Symbols/TypedConstant.cs [hunk 1]
```diff
@@ -104,7 +104,9 @@ public bool IsNull
         }
 
         /// <summary>
-        /// The value for a <see cref="TypedConstant"/> array. 
+        /// The value for a <see cref="TypedConstant"/> array.
+        /// Returns a default ImmutableArray if null was passed as the array value;
```
[the remark this answers is not in the document]
Comment (GiovanniBraconi) [reply src/Compilers/Core/Portable/Symbols/TypedConstant.cs]: Thanks for the suggestion 😄.

**Review by jaredpar — APPROVED** [review]

**Comment by jaredpar** [comment]
@333fred, @cston PTAL

File: src/Compilers/Core/Portable/Symbols/TypedConstant.cs [hunk 2]
```diff
@@ -104,7 +104,9 @@ public bool IsNull
         }
 
         /// <summary>
-        /// The value for a <see cref="TypedConstant"/> array. 
+        /// The value for a <see cref="TypedConstant"/> array.
+        /// Returns a default ImmutableArray if <see langword="null"/> was passed as the array value;
```
Comment (333fred) [inline src/Compilers/Core/Portable/Symbols/TypedConstant.cs:108]:
```suggestion
        /// Returns a <see langword="default" /> <c>ImmutableArray</c> if <see langword="null"/> was passed as the array value;
```

**Comment by jaredpar** [comment]
@333fred for a second list

**Review by 333fred — APPROVED** [review]

**Pull request merged**